bin/init.ml

```diff
@@ -131,13 +137,13 @@ let term =
     let doc =
       "A comma separated list of libraries on which the component depends"
     in
-    Arg.(value & opt (list component_name_conv) [] & info [ "libs" ] ~docv ~doc)
```

Comment:
With the current code, non-sense library names are properly guarded against:

```
dune init exe foo --libs bsadf+@adf,238sdfni
dune: option `--libs': invalid element in list (`bsadf+@adf,238sdfni'):
      invalid component name `bsadf+@adf'
      Library names must be non-empty and composed only of the
      following
      characters: 'A'..'Z', 'a'..'z', '_' or '0'..'9'.
Usage: dune init [OPTION]... INIT_KIND NAME [PATH]
Try `dune init --help' or `dune --help' for more information.
```

But removing this validation for the much weaker atom conversion allows this kind of thing without any protection.

Reply:
I think the right way to fix this would be to correct the `component_name_parser` so that it allows the correct format and rejects the incorrect ones, or perhaps to split into two separate converters: one for valid dune component names (leaving this as it is I guess), and another to validate library names, that allow `.` and `-`. Does this make sense?

@jeremiedimino, do you happen to know if there's a function in the code base that validates library names like `lwt.unix` and `lwt-unix`, but excludes strings with other characters in the mix?

Reply:
> I think the right way to fix this would be to correct the component_name_parser so that it allows the correct format and rejects the incorrect ones, or perhaps to split into two separate converters: one for valid dune component names (leaving this as it is I guess), and another to validate library names, that allow . and -. Does this make sense?

Yes, it sure does. 

I did try to find lib validation logic in the current dune codebase but wasn't successful. Additionally, I checked https://github.com/ocaml/ocamlfind source codebase to see if there a validation routine, but nope, it is just a string there too. Another possibility I toyed with was adding additional characters - mainly '.' and '-' - to the Lib_name.Local.t validation but I digress. 

The main issue is that in the current ocaml ecosystem it seems there isn't a formalized notion of what constitutes as a valid 'public' library name or that I am unaware of it. I suspect a valid library name corresponds to a valid directory name/filename but I digress. I also noticed that dune allows one to use ```/``` in the dune ```public_name``` attribute of the ```library/executable``` stanza. 

The closest of valid lib name that I have come across is ```Lib_name.t```, which is basically just ```Some s```, i.e. no validation done.

Reply:
> 
> 
> With the current code, non-sense library names are properly guarded against:
> 
> ```
> dune init exe foo --libs bsadf+@adf,238sdfni
> dune: option `--libs': invalid element in list (`bsadf+@adf,238sdfni'):
>       invalid component name `bsadf+@adf'
>       Library names must be non-empty and composed only of the
>       following
>       characters: 'A'..'Z', 'a'..'z', '_' or '0'..'9'.
> Usage: dune init [OPTION]... INIT_KIND NAME [PATH]
> Try `dune init --help' or `dune --help' for more information.
> ```
> 
> But removing this validation for the much weaker atom conversion allows this kind of thing without any protection.

Following from my comment above. I think the current ```Lib_name.Local``` validation for a public name  of a lib/executable is too narrow as it seems to reject valid library names being used in the ocaml ecosystem, eg. ```lwt.unix, js_of_ocaml-ppx,```.

Reply:
Perhaps, this ocaml RFC will formalize the library name - [ocamllib](https://github.com/ocaml/RFCs/blob/master/rfcs/ocamlib.md).

Reply:
Yeah, you're definitely right that the current validation is too narrow. I also looked for an existing validation function and didn't find one. It shouldn't be a big lift to write one, but I can do that later as a followup.

I think you're changes here fix something that is obviously a problem, where as my concerns are more defensive, and not obviously causing problem. So I'm OK with merging this is as (once the atom converter logic is cleaned up).

Thanks!